### PR TITLE
Pass computeDefaultStrings to query server when compiling queries

### PR DESCRIPTION
**Pull request by aeisenberg**
## Checklist

- [x] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [n/a] Issues have been created for any UI or other user-facing changes made by this pull request.
- [n/a] `@github/docs-content-dsp` has been cc'd in all issues for UI or other user-facing changes made by this pull request.

### CODE REVIEW

File: extensions/ql-vscode/src/vscode-tests/no-workspace/query-test.ts [hunk 1]
```diff
@@ -168,6 +168,7 @@ describe('using the query server', function() {
             localChecking: false,
             noComputeGetUrl: false,
             noComputeToString: false,
+            computeDefaultStrings: true
```
Comment (adityasharad) [inline extensions/ql-vscode/src/vscode-tests/no-workspace/query-test.ts:171]: Can you add a new `QueryTestCase` to exercise this?
The simplest I can think of is a query that looks like this:

```ql
newtype TUnit = MkUnit()

class Unit extends TUnit {
  Unit() { this = MkUnit() }

  string toString() { none() }
}

from Unit u
select u
```

Existing behaviour: this produces 0 results.
New behaviour: this produces 1 result, whose text is `(no string representation)`.

**Review by adityasharad — COMMENTED** [review]
Nice. I suggest adding a query integration test.

**Comment by adityasharad** [comment]
Worth adding a short changelog entry too, since it's actually user-visible. Something like "Query results without a string representation were previously not computed. They will now be computed and displayed with a default placeholder string."

File: extensions/ql-vscode/src/vscode-tests/no-workspace/query-test.ts (hunk 1 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/vscode-tests/no-workspace/query-test.ts:171]: I added that integration test to the list. #676

Currently, even the basic query test is failing. Need to work on that.

Comment (adityasharad) [reply extensions/ql-vscode/src/vscode-tests/no-workspace/query-test.ts:171]: Sure. For this particular test, we already have tests in `query-test` to run a simple query and get the raw results out, so I suggest adding a case in this PR if possible without needing the new integration test machinery.

Comment (aeisenberg) [reply extensions/ql-vscode/src/vscode-tests/no-workspace/query-test.ts:171]: Ah...I see. OK. I will add that tomorrow.

Comment (aeisenberg) [reply extensions/ql-vscode/src/vscode-tests/no-workspace/query-test.ts:171]: So, those tests have never been running in CI. Our scripts match on `*.test.ts`, but that file is `query-test.ts`. I will fix.

Comment (adityasharad) [reply extensions/ql-vscode/src/vscode-tests/no-workspace/query-test.ts:171]: 😱

**Comment by aeisenberg** [comment]
Test is added and and query tests are now running in CI.

File: extensions/ql-vscode/CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,7 @@
 ## [UNRELEASED]
 
 - Fix bug when removing databases where sometimes the source folder would not be removed from the workspace or the database files would not be removed from the workspace storage location. [#692](https://github.com/github/vscode-codeql/pull/692)
+- Always pass the `computeDefaultStrings` flag to query compilation commands. This will change the output of some queries. [#694](https://github.com/github/vscode-codeql/pull/694)
```
Comment (adityasharad) [inline extensions/ql-vscode/CHANGELOG.md:6]: I think I put this into an earlier comment I can't find - I'd prefer to mention what the user-facing change here is, i.e. query results with no string representation were previously omitted, and will now be displayed with placeholder text.

File: extensions/ql-vscode/src/vscode-tests/no-workspace/query.test.ts [hunk 3]
```diff
@@ -61,21 +59,27 @@ type QueryTestCase = {
...
   {
-    queryPath: path.join(__dirname, '../data/query.ql'),
+    queryPath: path.join(baseDir, 'query.ql'),
     expectedResultSets: {
       '#select': [[42, 3.14159, 'hello world', true]]
     }
   },
   {
-    queryPath: path.join(__dirname, '../data/multiple-result-sets.ql'),
+    queryPath: path.join(baseDir, 'compute-default-strings.ql'),
```
Comment (adityasharad) [inline extensions/ql-vscode/src/vscode-tests/no-workspace/query.test.ts:68]: Looks great! The test failures are now from `files.test.ts`, which expected to find only 2 queries in this folder and now found the third one.

File: extensions/ql-vscode/CHANGELOG.md (hunk 2 above)
Comment (aeisenberg) [reply extensions/ql-vscode/CHANGELOG.md:6]: Sure. I didn't quite know how to say it. This sounds good.

File: extensions/ql-vscode/src/vscode-tests/no-workspace/query.test.ts (hunk 3 above)
Comment (aeisenberg) [reply extensions/ql-vscode/src/vscode-tests/no-workspace/query.test.ts:68]: Fixed now. Also, the sort was wrong also.

**Review by adityasharad — APPROVED** [review]

**Pull request merged**